### ISSUE TITLE
[a11y]:Text for icons should not include information about variations on an icon

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#10179]
### Package

carbon-components

### Browser

Chrome

### Operating System

MacOS

### Package version

carbondesignsystem.com

### Automated testing tool and ruleset

n/a

### Assistive technology

n/a

### Description

There are variations in icons in the [Carbon icon library ](https://www.carbondesignsystem.com/guidelines/icons/library/)that should not have the variation added as part of the accessible name. 

The words include:
![image](https://user-images.githubusercontent.com/14298245/143623917-12ad4401-eb04-41bf-97af-963ebe900680.png)
^ "filled" (to distinguish between a filled and unfilled version of the same icon

![image](https://user-images.githubusercontent.com/14298245/143624055-510c6e56-95cf-4e3a-ab63-7ea98da3edde.png)
^ "outlined" (to similarly distinguish between a filled and outlined version of an icon)

![image](https://user-images.githubusercontent.com/14298245/143624495-9dbc4ab7-054e-48e3-8905-3b0e168d644e.png)
^ "alt" (to identify an alternative version of the same icon)

![image](https://user-images.githubusercontent.com/14298245/143624419-861114a7-e194-4cd5-a1f7-f2d612b567e4.png)
^ "solid" (basically when only a symbol in the icon is filled)

![image](https://user-images.githubusercontent.com/14298245/143624678-9a8cae4d-b900-47fe-84fe-79061d2d0643.png)
^ "1" and other numbering, to designate multiple variations in an icon

![image](https://user-images.githubusercontent.com/14298245/143624928-8f4cc3e0-d895-4f25-8f13-446329f8234d.png)
^ "inverted" (where the shape has been flipped)

For example, there are 55 icons in Carbon that are variations on an icon that contain the word "alt" (to designate that they are alternative versions)

While this is a useful label in the context of a list of icons in the icon library, it is meaningless from an accessible alternative text perspective. Unfortunately, in every case, the visible icon label in the list of icons is also added as the title of the svg icon.
![image](https://user-images.githubusercontent.com/14298245/143623672-c7351188-301a-4e7b-a52b-2cd8be8d8d68.png)

The exception to this is where the filled state may contain meaning. This is very rare -- and in the case where the fill is used to contain meaning, the alt would likely be better to reflect the meaning, than the visual state. For instance, in the following images, stars and thumbs up icons are both filled and unfilled. These are often used to indicate selection. A better alt would be  "star unselected" and "star selected". It's also possible that when they are used this way, the element's state is controlled programmatically (for instance up and down chevrons to indicate expanded and collapsed) and the SVG title is is superfluous.
![image](https://user-images.githubusercontent.com/14298245/143625059-b8b5f2c2-e042-48a1-9731-b26a9d34082f.png)

**Important note:** the icons in this library have been implemented in a peculiar way:
1. they have been given an `aria-hidden="true"`
2. they have not been given a `role="image"`

As a result, they are not identified by assistive technologies. However, my assumption is that when implemented, the icons will _not_ be aria-hidden, and that they will be assigned a role of image, and the title value of the SVG will be accessible to most combinations of AT and browser ([although the behaviour may vary](https://www.deque.com/blog/creating-accessible-svgs/))

The solution to this will be to remove these words from the title name, where the implementation allows the title to be detected as the accessible name (i.e., not hidden, role is given as img)

### WCAG 2.1 Violation

1.1.1 Non-text Content

### CodeSandbox example

https://www.carbondesignsystem.com/guidelines/icons/library/

### Steps to reproduce

1. Go to https://www.carbondesignsystem.com/guidelines/icons/library/
2. Do a search for " alt"
3. Inspect the code to validate the svg title contains "alt"

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by jnm2377** [comment]
Hi @mbgower thanks for opening this issue. I'm not sure I understand why you're saying this isn't accessible. It seems like the issue is just with the naming, but all of our icons have (and need) different names. For example, the first set of icons you showed: `Continue` and `Continue filled`, the title reflects the actual name of the icon components `<Continue/>` and `<ContinueFilled/>`. It makes sense to me that the title of the icon card would be what the component name is bc users know and expect to use that name.

**Comment by jnm2377** [comment]
Also, is this an issue with the iconography cards/page in the website, or the actual icons (i.e. using the svg/components)? If the former, then this issue should be transferred to the website repo.

**Comment by joshblack** [comment]
> Unfortunately, in every case, the visible icon label in the list of icons is also added as the title of the svg icon.

This is only used in the context of the icon library page and does not appear in production code 👍 

> the icons in this library have been implemented in a peculiar way

The attributes of the SVG files follow a handful of rules that a developer can control:

- if `aria-label`, `aria-labelledby`, or `title` is set, then apply `role="img"`
  - If tabindex is defined, add `focusable="true"` and `tabindex` property
  - Else, add `aria-hidden="true"`

Let me know if you would like changes to this heuristic or if your looking for changes on the website itself 👀
